src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -92,6 +97,25 @@ private static bool IsValidForTaskLikeTypeOnlyContext(ISymbol symbol, TSyntaxCon
             return symbol.IsAwaitableNonDynamic(context.SemanticModel, context.Position);
         }
 
+        private static bool IsValidForGenericConstraintContext(ISymbol symbol)
+        {
+            if (symbol.IsNamespace() ||
+                symbol.IsKind(SymbolKind.TypeParameter))
+            {
+                return true;
+            }
+
+            if (symbol is not INamedTypeSymbol namedType ||
+                symbol.IsDelegateType() ||
+                namedType.IsEnumType() ||
+                namedType.IsStructType())
+            {
+                return false;
+            }
+
+            return !(namedType.IsStatic || namedType.IsSealed);
```

Comment:
i'm wary about things like the `IsSealed` check.  if i do `where T : SomeSealedType.SomeInterface` that should work.